protostar/cheatable_starknet/callable_hint_locals/declare_cairo0_hint_local.py

```diff
@@ -38,11 +43,17 @@ def _build(self) -> Any:
 
     def declare_cairo0(self, contract: CairoShortString) -> DeclaredContract:
         contract_str = short_string_to_str(contract)
-        compiled_contract = (
-            self._project_compiler.compile_contract_from_contract_identifier(
-                contract_str
+        try:
+            compiled_contract = (
+                self._project_compiler.compile_contract_from_contract_identifier(
+                    contract_str
+                )
             )
-        )
+        except ContractNameNotFoundException as err:
+            raise CheatcodeException(self, err.message) from err
+        except ParserError as err:
```

Comment:
Not sure why are we catching `ParserError` here. `ProjectCompiler` should throw `CompilationException` if there's a problem with compilation (though the catching logic seems incomplete there so we need some additional catching/updating it anyways).

Reply:
it seems like a bug here

Reply:
This will be tightly coupled with this change so I will fix this bug in this PR